crates/uv/src/lib.rs

```diff
@@ -1225,21 +1225,23 @@ async fn run(mut cli: Cli) -> Result<ExitStatus> {
                     .combine(Refresh::from(args.settings.resolver.upgrade.clone())),
             );
 
-            let mut requirements = Vec::with_capacity(
-                args.with.len() + args.with_editable.len() + args.with_requirements.len(),
-            );
-            for package in args.with {
-                requirements.push(RequirementsSource::from_with_package_argument(&package)?);
+            let with_executables_from = args
+                .with_executables_from
+                .iter()
+                .map(|pkg| RequirementsSource::from_package(pkg))
+                .collect::<Result<Vec<_>, _>>()?;
+
```

Comment:
I think we probably need some sort of comment here that makes it clear that `with_executables_from` are being added to `with`. It's easy to miss otherwise.